app/src/main/java/com/example/masala_food_recipes/data/interactors/Cuisine.kt

```diff
@@ -0,0 +1,23 @@
+package com.example.masala_food_recipes.data.interactors
+
+import com.example.masala_food_recipes.data.entities.Recipe
+
+class Cuisine(private val recipes:List<Recipe>) {
+
+    fun getCuisines():Map<String,Int>{
+        val mapCuisine = mutableMapOf<String,Int>()
+        recipes.forEach {
```

Comment:
Your code works correct But you can search about groupBy, mapValues try to simplify your code